### PR TITLE
MULE-17921: Support Platform Managed OAuth tokens

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/OAuthService.java [hunk 1]
```diff
@@ -46,16 +47,35 @@
...
    * @param expressionEvaluator the object used to evaluate expressions.
-   * 
    * @return a builder for an authorization-code grant type dancer.
    */
   <T> OAuthAuthorizationCodeDancerBuilder authorizationCodeGrantTypeDancerBuilder(LockFactory lockProvider,
                                                                                   Map<String, T> tokensStore,
                                                                                   MuleExpressionLanguage expressionEvaluator);
 
+  /**
+   * Creates a builder for an {@link PlatformManagedOAuthDancer} that obtains access tokens that are provisioned and managed
```
Comment (elrodro83) [inline modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/OAuthService.java]: an -> a

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/PlatformManagedOAuthDancer.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * <p>
+ * Since the authorizations are performed by the platform, this dancer remains agnostic of the grant type that was used to
+ * obtain them.
+ * <p>
+ * Platform Managed OAuth is an experimental feature. It will only be enabled on selected environments and scenarios.
+ * Backwards compatibility is not guaranteed.
+ *
+ * @since 4.3.0
+ */
+@NoImplement
```
Comment (elrodro83) [inline modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/PlatformManagedOAuthDancer.java]: won't this implementation be in the service artifact?

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+ * Because the actual OAuth authorization is not performed by Mule but the Anypoint Platform, some of the methods in the base
+ * {@link OAuthDancerBuilder} do not apply and will throw an {@link UnsupportedOperationException}. The the javadocs of all
+ * methods defined and overridden in this interface to know which ones are valid.
+ * <p>
+ * Platform Managed OAuth is an experimental feature. It will only be enabled on selected environments and scenarios.
+ * Backwards compatibility is not guaranteed.
+ *
+ * @since 4.3.0
+ */
+@NoImplement
```
Comment (elrodro83) [inline modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java]: won't this implementation be in the service artifact?

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/listener/PlatformManagedOAuthStateListener.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Allows to get notified about certain events related to Platform Managed OAuth tokens
+ * <p>
+ * Platform Managed OAuth is an experimental feature. It will only be enabled on selected environments and scenarios. Backwards compatibility is
+ * not guaranteed.
+ *
+ * @since 4.3.0
+ */
+@Experimental
+@NoImplement
```
Comment (elrodro83) [inline modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/listener/PlatformManagedOAuthStateListener.java]: won't this implementation be in the service artifact?

**Comment by elrodro83** [comment]
@eze210 please review the http-api part

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/PlatformManagedOAuthDancer.java (hunk 2 above)
Comment (marianogonzalez) [reply modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/PlatformManagedOAuthDancer.java]: Yes, on the EE one.

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java (hunk 3 above)
Comment (marianogonzalez) [reply modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java]: on the EE one

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/listener/PlatformManagedOAuthStateListener.java (hunk 4 above)
Comment (marianogonzalez) [reply modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/listener/PlatformManagedOAuthStateListener.java]: yes, on the EE one

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+   * @return {@code this} builder
+   */
+  OAuthPlatformManagedDancerBuilder connectionId(String connectionId);
+
+  /**
+   * Adds the {@code listener}. Listeners will be invoked in the same order as they were added
+   *
+   * @param listener the {@link PlatformManagedOAuthStateListener} to be added
+   * @throws IllegalArgumentException if the {@code listener} is {@code null}
+   * @since 4.2.1
```
Comment (eze210) [inline modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java]: 4.2.1 or 4.3.0?

File: modules/http-api/src/main/java/org/mule/runtime/http/api/client/ManagedLifecycleHttpClient.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Decorates a {@link HttpClient} so that invocations to {@link HttpClient#start()} and {@link HttpClient#stop()} have no effect.
+ * <p>
+ * This is useful (and recommended) for cases in which a given component owns an {@link HttpClient} but needs to expose said
+ * client to other consumers. Since hte lifecycle of the client should only be handled by the owner, it is good practice to use
```
Comment (eze210) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/client/ManagedLifecycleHttpClient.java]: hte -> the

**Pull request merged**